app/(tabs)/clubs.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState } from 'react';
+import { useForm, Controller, SubmitHandler } from 'react-hook-form';
+import { useRouter } from "expo-router";
+import FontAwesome from '@expo/vector-icons/FontAwesome';
+import { useFocusEffect } from '@react-navigation/native';
+import { Input, InputField, InputIcon, InputSlot } from '@/components/ui/input';
+import { useQuery } from '@tanstack/react-query';
+import { fetchClubByName } from '../../api/clubService';
+import CardLayout from '@/components/cards/CardLayout';
+import { SimplifiedClubInfo } from '../../types/ClubInfoTypes';
+
+const { width: screenWidth } = Dimensions.get('window');
+
+type FormData = {
+  searchInput: string;
+};
+
+export default function Clubs() {
+  const router = useRouter();
+  const { control, handleSubmit, formState: { errors }, reset } = useForm<FormData>();
+  const [inputText, setInputText] = useState('');
+  const { data: clubs, isLoading, isError } = useQuery<[SimplifiedClubInfo], Error>(
+    {
+      queryKey: ['clubSearch', inputText],
+      queryFn: () => fetchClubByName(inputText),
+      enabled: !!inputText,
+    }
+  );
+
+  // Clears form state when screen gains focus
+  useFocusEffect(
+    React.useCallback(() => {
+      reset();
+      setInputText('');
+    }, [reset])
+  );
+
+  const handleSearchClick: SubmitHandler<FormData> = (data) => {
+    setInputText(data.searchInput);
+  };
+
+  const handleClubClicked = (clubId: string) => {
+    router.push(`/clubDetails?id=${clubId}`);
+  }
+
+  return (
+    <View style={styles.container}>
+      <ScrollView contentContainerStyle={styles.scrollContainer} showsVerticalScrollIndicator={false}>
+        <Controller
+          control={control}
+          name="searchInput"
+          defaultValue=""
+          rules={{ required: 'Must insert a value' }}
+          render={({ field: { onChange, onBlur, value } }) => (
+            <>
+              <Input
+                variant="rounded"
+                size="md"
+                style={{ borderColor: '#ffffff', margin: 16 }}
+              >
+                <InputField
+                    placeholder="Enter club name..."
+                    style={{ color: '#ffffff' }}
+                    onChangeText={(text) => {
+                      onChange(text);
+                    }}
+                    onBlur={onBlur}
+                    value={value}
+                />
+                <InputSlot>
+                    <TouchableOpacity onPress={handleSubmit(handleSearchClick)}>
+                    <InputIcon as={() => <FontAwesome name="search" size={20} color="#ffffff" style={{ padding: 16 }} />} />
+                    </TouchableOpacity>
+                </InputSlot>
+              </Input>
+              {errors.searchInput && (
+                <Text style={styles.errorText}>{errors.searchInput.message}</Text>
+              )}
+              
+              {isLoading && <Text style={styles.text}>Loading...</Text>}
```

Comment:
This is just a personal preference but i would prefer to do an early return on the component

`
if (isLoading) {
  return (<Text>Loading...</Text>)
}
`